cmd/soroban-rpc/internal/methods/get_transactions.go

```diff
@@ -0,0 +1,230 @@
+	// ResultMetaXdr is the TransactionMeta XDR value.
+	ResultMetaXdr string `json:"resultMetaXdr"`
+	// DiagnosticEventsXDR is present only if transaction was not successful.
+	// DiagnosticEventsXDR is a base64-encoded slice of xdr.DiagnosticEvent
+	DiagnosticEventsXDR []string `json:"diagnosticEventsXdr,omitempty"`
+	// Ledger is the sequence of the ledger which included the transaction.
+	Ledger uint32 `json:"ledger"`
+	// LedgerCloseTime is the unix timestamp of when the transaction was included in the ledger.
+	LedgerCloseTime int64 `json:"createdAt"`
+}
+
+// GetTransactionsResponse encapsulates the response structure for getTransactions queries.
+type GetTransactionsResponse struct {
+	Transactions          []transactionInfo `json:"transactions"`
+	LatestLedger          uint32            `json:"latestLedger"`
+	LatestLedgerCloseTime int64             `json:"latestLedgerCloseTimestamp"`
+	OldestLedger          uint32            `json:"oldestLedger"`
+	OldestLedgerCloseTime int64             `json:"oldestLedgerCloseTimestamp"`
+	Cursor                string            `json:"cursor"`
+}
+
+type transactionsRPCHandler struct {
+	ledgerReader      db.LedgerReader
+	dbReader          db.TransactionReader
+	maxLimit          uint
+	defaultLimit      uint
+	logger            *log.Entry
+	networkPassphrase string
+}
+
+// getTransactionsByLedgerSequence fetches transactions between the start and end ledgers, inclusive of both.
+// The number of ledgers returned can be tuned using the pagination options - cursor and limit.
+func (h transactionsRPCHandler) getTransactionsByLedgerSequence(ctx context.Context, request GetTransactionsRequest) (GetTransactionsResponse, error) {
+	ledgerRange, err := h.dbReader.GetLedgerRange(ctx)
+	if err != nil {
+		return GetTransactionsResponse{}, &jrpc2.Error{
+			Code:    jrpc2.InvalidParams,
+			Message: err.Error(),
+		}
+	}
+
+	err = request.isValid(h.maxLimit, ledgerRange)
+	if err != nil {
+		return GetTransactionsResponse{}, &jrpc2.Error{
+			Code:    jrpc2.InvalidParams,
+			Message: err.Error(),
+		}
+	}
+
+	// Move start to pagination cursor
+	start := toid.New(int32(request.StartLedger), 1, 1)
+	limit := h.defaultLimit
+	if request.Pagination != nil {
+		if request.Pagination.Cursor != nil {
+			start = request.Pagination.Cursor
+			// increment tx index because, when paginating,
+			// we start with the item right after the cursor
+			start.TransactionOrder++
+		}
+		if request.Pagination.Limit > 0 {
+			limit = request.Pagination.Limit
+		}
+	}
+
+	// Iterate through each ledger and its transactions until limit or end range is reached.
+	// The latest ledger acts as the end ledger range for the request.
+	var txns []transactionInfo
+	var cursor *toid.ID
+LedgerLoop:
+	for ledgerSeq := start.LedgerSequence; ledgerSeq <= int32(ledgerRange.LastLedger.Sequence); ledgerSeq++ {
+		// Get ledger close meta from db
+		ledger, found, err := h.ledgerReader.GetLedger(ctx, uint32(ledgerSeq))
```

Comment:
This is a thought for the future: we should probably do something more like `getLedgers()` and then process ledgers in batches so that we don't have to have one SQL query for each iteration (while still respecting `limit`, of course, but a little extra memory is worth saving queries).

We can include this as part of (or after, at the very least) the `getLedgers` work in #111.

Reply:
@Shaptic Yeah I agree. We should do the batch processing as its more efficient.